apps/storefront/src/components/ui/B2BSwitchCompanyModal.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box } from '@mui/material';
+import Cookies from 'js-cookie';
+
+import B3Dialog from '@/components/B3Dialog';
+import useMobile from '@/hooks/useMobile';
+import { endUserMasqueradingCompany, startUserMasqueradingCompany } from '@/shared/service/b2b';
+import { deleteCart } from '@/shared/service/bc/graphql/cart';
+import { store, useAppSelector } from '@/store';
+import { setCompanyHierarchyInfoModules } from '@/store/slices/company';
+import { setCartNumber } from '@/store/slices/global';
+import b2bLogger from '@/utils/b3Logger';
+import { deleteCartData } from '@/utils/cartUtils';
+
+interface B2BSwitchCompanyModalPropsTypes {
+  open: boolean;
+  title: string;
+  fullWidth?: boolean;
+  tipText: string;
+  setIsOpenSwitchCompanyModal: (value: boolean) => void;
+  switchCompanyId: string | number | undefined;
+  rightSizeBtn?: string;
+}
+
+function B2BSwitchCompanyModal(props: B2BSwitchCompanyModalPropsTypes) {
+  const {
+    open,
+    title,
+    fullWidth = true,
+    tipText,
+    setIsOpenSwitchCompanyModal,
+    switchCompanyId = 0,
+    rightSizeBtn = '',
+  } = props;
+  const b3Lang = useB3Lang();
+  const [isMobile] = useMobile();
+
+  const { id: currentCompanyId } = useAppSelector(({ company }) => company.companyInfo);
+
+  const { companyHierarchyList } = useAppSelector(({ company }) => company.companyHierarchyInfo);
+
+  const [loading, setLoading] = useState(false);
+
+  const handleClose = () => {
+    setIsOpenSwitchCompanyModal(false);
+  };
+
+  const handleSwitchCompanyClick = async () => {
+    setLoading(true);
+    try {
+      if (+switchCompanyId === +currentCompanyId) {
+        await endUserMasqueradingCompany();
+      } else if (switchCompanyId) {
+        await startUserMasqueradingCompany(+switchCompanyId);
+      }
+
+      const cartEntityId = Cookies.get('cartId');
+      if (cartEntityId) {
+        const deleteCartObject = deleteCartData(cartEntityId);
+
+        await deleteCart(deleteCartObject);
+
+        store.dispatch(setCartNumber(0));
+      }
+
+      if (switchCompanyId) {
+        store.dispatch(
+          setCompanyHierarchyInfoModules({
+            selectCompanyHierarchyId:
+              +switchCompanyId === +currentCompanyId ? '' : +switchCompanyId,
+            companyHierarchyList: companyHierarchyList || [],
+          }),
+        );
+      }
+    } catch (error) {
+      b2bLogger.error(error);
+    } finally {
+      setLoading(false);
+      handleClose();
+    }
+  };
+
+  return (
+    <B3Dialog
+      isOpen={open}
+      rightSizeBtn={rightSizeBtn || b3Lang('global.B2BSwitchCompanyModal.confirm.button')}
+      title={title || b3Lang('global.B2BSwitchCompanyModal.title')}
+      fullWidth={fullWidth}
+      maxWidth={false}
+      loading={loading}
+      handleLeftClick={handleClose}
+      handRightClick={handleSwitchCompanyClick}
+      dialogSx={{
+        '& .MuiPaper-elevation': {
+          width: isMobile ? '100%' : `480px`,
```

Comment:
Shouldn't this be part of the B3Dialog logic/component prop?

Why is the consumer of the B3Dialog having to set the width via a style prop?

Reply:
Thank you for your feedback!  You're absolutely right;  it makes sense for the B3Dialog component to manage its own width logic based on props, rather than having the consumer handle this via the dialogSx prop.

I'll update the B3Dialog component to include a prop (e.g., dialogWidth) that allows specifying widths.